### PR TITLE
Fix panic on invalid file URI #3275

### LINKED ISSUE #3275: Crash parsing invalid file URI
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same problem, before opening a new one.
- [X] This issue only relates to a single bug. I will open new issues for any other problems.

### Describe the bug

We used the Fuzz engine to modify some Test(TestFromJSON) data, and then the following crash appeared. We hope to get the help of the developer to confirm whether it is a real bug.

### How to reproduce

The simplified test code is in **Example code**

### Screenshots

crash log:
```
panic: runtime error: slice bounds out of range [5:4] [recovered]
        panic: runtime error: slice bounds out of range [5:4]

goroutine 35 [running]:
testing.tRunner.func1.2({0x62e6c0, 0xc0001d43d8})
        /home/zjx/.local/go/src/testing/testing.go:1211 +0x24e
testing.tRunner.func1()
        /home/zjx/.local/go/src/testing/testing.go:1214 +0x218
panic({0x62e6c0, 0xc0001d43d8})
        /home/zjx/.local/go/src/runtime/panic.go:1038 +0x215
fyne.io/fyne/v2/storage/repository.ParseURI({0xc0001b9770, 0x4})
        /home/zjx/workspace/gowork/src/purelib/fyne/storage/repository/parse.go:63 +0x737
fyne.io/fyne/v2/storage.ParseURI(...)
        /home/zjx/workspace/gowork/src/purelib/fyne/storage/uri.go:35
fyne.io/fyne/v2/theme.uriString.resource({0xc0001b9770, 0xc00019a128})
        /home/zjx/workspace/gowork/src/purelib/fyne/theme/json.go:82 +0x1f
fyne.io/fyne/v2/theme.(*jsonTheme).Font(0xc0001b0528, {0x0, 0x0, 0x1, 0x0, 0x0})
        /home/zjx/workspace/gowork/src/purelib/fyne/theme/json.go:146 +0xbb
fyne.io/fyne/v2/theme.FuzzTestFromJSON({0xc00036f600, 0x22, 0x200})
        /home/zjx/workspace/gowork/src/purelib/fyne/theme/json_test.go_TestFromJSON_test.go:74 +0x143
```

### Example code

```go
        var1:="0000"
	var2:=`{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"Fonts":{"monospace":"file"}}0000{"F`
	repository.Register(var1, intRepo.NewFileRepository())
	th, err := FromJSON(var2)
	assert.Nil(t, err)
	th.Font(fyne.TextStyle{Monospace: true}).Name()
```

### Fyne version

638ae24e4bebfc0b6dbaabf8db5807eeabfafc61

### Go compiler version

1.17.8

### Operating system

Linux

### Operating system version

Linux r920 4.15.0-159-generic #167-Ubuntu

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by ChandanChainani**
### Description:

Fixes #3275

### Checklist:

- [x] Tests included.
- [x] Lint and formatter run with no errors.
- [ ] Tests all pass.

### CODE REVIEW

**Comment by Jacalz** [comment]
Thanks for the PR. You have created the PR against the master branch but the PR template states that it should target `develop`. Would you mind changing the target branch to be correct?

**Comment by Jacalz** [comment]
Also, please add "Fixes #3275" to the description. This will close the issue automatically when the PR is merged

**Comment by ChandanChainani** [comment]
@Jacalz Thanks, updated as per comments

File: storage/repository/parse.go [hunk 1]
```diff
@@ -60,13 +61,16 @@ func ParseURI(s string) (fyne.URI, error) {
...
+			if len(path) > 2 && path[:2] == "//" {
+				path = path[2:]
+			}
 
-		// Windows files can break authority checks, so just return the parsed file URI
-		return NewFileURI(path), nil
+			// Windows files can break authority checks, so just return the parsed file URI
+			return NewFileURI(path), nil
+		}
+		return nil, errors.New("not a valid URI")
```
Comment (Jacalz) [inline storage/repository/parse.go:73]: I would prefer if you check that the code path is invalid and error out in this case. In other words, you keep the code indented as it was before and then have the error handing in an if-statement instead.
```go
if len(s) < 6 {
   return nil, errors.New("not a valid URI")
}
```

**Review by Jacalz — COMMENTED** [review]
It would also be good if you could rebase on `develop` given that https://github.com/fyne-io/fyne/pull/3301 landed recently and fixed another crash.

**Review by Jacalz — CHANGES_REQUESTED** [review]
I feel that it would be good to have an actual test case for `Parse` (in the right place) that tests when an error should happen. The new code path is never tested to actually be working unfortunately

**Review by andydotxyz — APPROVED** [review]
Looks good thanks

**Comment by andydotxyz** [comment]
@Jacalz there is still an outstanding change request from you but I think it's good now.

**Review by Jacalz — APPROVED** [review]
Indeed. Looks good. Thanks for working on this :)

**Pull request merged**